### ISSUE TITLE
Feature Request: Conditional Build Skipping

### ISSUE THREAD

**Issue by sinahwz** [buildkite/agent#725]
Hi, It would be great to have more customizable Build skipping scenarios. One simple scenario for example:
Don't cancel Builds that are running for a certain amount of time.
I personally would like to have my pipelines complete what they are doing, and not be canceled by a new push, just right before they are finished (in my case, between 25 to 30 minutes).

**Comment by lox** [comment]
Yup, that would be neat. We've always imagined that one day there would be a `buildkite-agent skip`, or perhaps a hook that ran before an agent accepted the job that allowed it to be skipped.

**Comment by href** [comment]
👍 We would like to skip building commits which have been tagged (which results in the same builds being run twice). We currently skip it by altering the pipeline steps, but ideally we could completely ignore those builds.

`buildkite-agent skip` sounds like that would be great for us.
